src/Rector/Class_/AddMethodCallToConstructorForClassesUsingTraitRector.php

```diff
@@ -0,0 +1,182 @@
+<?php
+declare(strict_types=1);
+
+namespace Sylius\SyliusRector\Rector\Class_;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Expression;
+use Rector\Compatibility\NodeFactory\ConstructorClassMethodFactory;
+use Rector\Core\Contract\Rector\ConfigurableRectorInterface;
+use Rector\Core\NodeManipulator\ClassInsertManipulator;
+use Rector\Core\NodeManipulator\ClassManipulator;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\ValueObject\MethodName;
+use Sylius\SyliusRector\NodeManipulator\ClassInterfaceManipulator;
+use Symplify\RuleDocGenerator\Exception\PoorDocumentationException;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+use function RectorPrefix202208\dump;
+
+/**
+ * @see \Sylius\SyliusRector\Tests\Rector\Class_\AddMethodCallToConstructor\AddMethodCallToConstructorTest
+ */
+final class AddMethodCallToConstructorForClassesUsingTraitRector extends AbstractRector implements ConfigurableRectorInterface
+{
+    private array $configuration = [];
+
+    public function __construct(
+        private ConstructorClassMethodFactory $constructorClassMethodFactory,
+        private ClassInsertManipulator $classInsertManipulator,
+        private ClassManipulator $classManipulator,
+    ) {
+    }
+
+    /**
+     * @throws PoorDocumentationException
+     */
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Adds given method calls to constructor for classes using given trait',
+            [
+                new CodeSample(
+                    <<<CODE_SAMPLE
+                use Sylius\Component\Channel\Model\Channel as BaseChannel;
+                
+                class Channel extends BaseChannel
+                {
+                }
+                CODE_SAMPLE,
+                    <<<CODE_SAMPLE
+                use Sylius\Component\Channel\Model\Channel as BaseChannel;
+                
+                class Channel extends BaseChannel implements \Sylius\MultiStorePlugin\BusinessUnits\Domain\Model\ChannelInterface
+                {
+                }
+                CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    public function configure(array $configuration): void
+    {
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * @param Class_ $node
+     * @throws \Exception
+     */
+    public function refactor(Node $node): Node|array|null
+    {
+        $newConstructorStmts = [];
```

Comment:
`Stmts`? Can you expand the shortcut?

Reply:
`Stmt` stands probably for `Statement` but everywhere in Rector it is called `Stmt` so to stay clear that I mean Rector's `Stmt` I decided to name it `Stmt`.

Reply:
I would be for not using shortcuts, but it could be optionally refactored in the separate PR